src/agent0/chainsync/db/hyperdrive/chain_to_db.py

```diff
@@ -100,3 +113,325 @@ def data_chain_to_db(interface: HyperdriveReadInterface, block: BlockData, sessi
 
     block_pool_info = convert_pool_info(pool_info_dict)
     add_pool_infos([block_pool_info], session)
+
+
+def _event_data_to_dict(in_val: EventData) -> dict[str, Any]:
+    out = dict(in_val)
+    # The args field is also an attribute dict, change to dict
+    out["args"] = dict(in_val["args"])
+
+    # Convert transaction hash to string
+    out["transactionHash"] = in_val["transactionHash"].to_0x_hex()
+    # Get token id field from args.
+    # This field is `assetId` for open/close long/short
+    return out
+
+
+# TODO cleanup
```

Comment:
This function is quite a beast, and has some mission-critical components. I think we should prioritize cleanup **and** testing as soon as this refactor is done.